onnxscript/test/function_libs/torch_aten/ops_correctness_test.py

```diff
@@ -0,0 +1,227 @@
+"""Test op correctness by comparing with PyTorch results."""
+from __future__ import annotations
+
+import copy
+import dataclasses
+import unittest
+from typing import Callable, Collection, Iterable, Optional, Sequence
+
+import numpy as np
+import torch
+from torch.testing._internal import common_device_type, common_methods_invocations
+from torch.testing._internal.opinfo import core as opinfo_core
+
+import onnxscript
+from onnxscript.fuction_libs.torch_aten.ops import core as core_ops
+
+SUPPORTED_DTYPES = (
+    # Boolean
+    torch.bool,
+    # Integers
+    torch.uint8,
+    torch.int8,
+    torch.int16,
+    torch.int32,
+    torch.int64,
+    # Floating types
+    torch.float16,
+    torch.float32,
+    torch.float64,
+)
+
+# Convenience tuples for creating dtype lists when skipping or xfailing tests
+
+BOOL_TYPES = (torch.bool,)
+
+INT_TYPES = (
+    torch.int8,
+    torch.int16,
+    torch.int32,
+    torch.int64,
+    torch.uint8,
+)
+
+FLOAT_TYPES = (
+    torch.float16,
+    torch.float32,
+    torch.float64,
+)
+
+
+@dataclasses.dataclass
+class DecorateMeta:
+    """A dataclass for storing information about a test case to skip or xfail.
+
+    Adapted from functorch: functorch/test/common_utils.py
+    """
+
+    op_name: str
+    variant_name: str
+    decorator: Callable
+    dtypes: Optional[Collection[torch.dtype]]
+    reason: str
+
+
+def xfail(
+    op_name: str,
+    variant_name: str = "",
+    *,
+    dtypes: Optional[Collection[torch.dtype]] = None,
+    reason: Optional[str] = None,
+):
+    """Expects an OpInfo test to fail.
+
+    Args:
+        op_name: The name of the operator.
+        variant_name: Optional OpInfo variant_test_name.
+        dtypes: The dtypes to expect the failure.
+        reason: The reason for the failure.
+    """
+    if reason is None:
+        raise ValueError("Please specify a reason.")
+    return DecorateMeta(
+        op_name=op_name,
+        variant_name=variant_name,
+        decorator=unittest.expectedFailure,
+        dtypes=dtypes,
+        reason=reason,
+    )
+
+
+def skip(
+    op_name: str,
+    variant_name: str = "",
+    *,
+    dtypes: Optional[Collection[torch.dtype]] = None,
+    reason: Optional[str] = None,
+):
+    """Skips an OpInfo test.
+
+    Args:
+        op_name: The name of the operator.
+        variant_name: Optional OpInfo variant_test_name.
+        dtypes: The dtypes to skip.
+        reason: The reason for skipping.
+    """
+    if reason is None:
+        raise ValueError("Please specify a reason.")
+    return DecorateMeta(
+        op_name=op_name,
+        variant_name=variant_name,
+        decorator=unittest.skip(f"Don't care: {reason}"),
+        dtypes=dtypes,
+        reason=reason,
+    )
+
+
+def add_decorate_info(
+    all_opinfos: Sequence[opinfo_core.OpInfo],
+    test_class_name: str,
+    base_test_name: str,
+    skip_or_xfails: Iterable[DecorateMeta],
```

Comment:
Could this be optional, or the default value could be None?

Reply:
skip_or_xfails is always required. Empty list if nothing to skip. Thoughts?

Reply:
Either an empty list or None works for me.

Reply:
Going for empty list